component/registration_form.jsx

```diff
@@ -0,0 +1,44 @@
+import { TYPE_JSON } from '../api/defs'
+
+export const registrationForm = () => {
+  const onSubmit = async (e) => {
+    e.preventDefault()
+    // form data
+    const form = new FormData(e.target)
+    const data = Object.fromEntries(form)
+    const json = JSON.stringify(data)
+    const fetching = await fetch('/api/user/registerForm', {
+      method: 'POST',
+      headers: { ...TYPE_JSON },
+      body: { ...json },
+    })
+    const response = await fetching.json()
+    if (response.status) {
+      // redirect some where
```

Comment:
this is exactly why I do not want to have `status` as a property here, I found this code confusing because:
- You called the `Response` object from `fetch` `fetching`
- You named the actual body `response` and read it's `status` property
while the original `fetch` response has a status, that is a number `fetching.status` in that case

just check if the errors property exist:
```js
    const response = await fetch('/api/user/registerForm', {
      method: 'POST',
      headers: { ...TYPE_JSON },
      body: { ...json }, // <- this is bad btw, why spread a string into an object ?
    })
    const body = await response.json()
    if (body?.errors.length) {
      // handle errors
    } else {
      // continue...
    }
```